SMLHelper/Json/ConfigFile.cs

```diff
@@ -0,0 +1,90 @@
+﻿namespace SMLHelper.V2.Json
+{
+    using System.IO;
+    using System.Reflection;
+    using System.Linq;
+    using SMLHelper.V2.Json.Converters;
+    using SMLHelper.V2.Json.ExtensionMethods;
+    using SMLHelper.V2.Json.Interfaces;
+    using Oculus.Newtonsoft.Json;
+    using Oculus.Newtonsoft.Json.Converters;
+
+    /// <summary>
+    /// A simple implementation of <see cref="IJsonFile"/> for use with config files.
+    /// </summary>
+    public abstract class ConfigFile : IJsonFile
```

Comment:
Should this class really be abstract?

Reply:
That's the entire point of it.  
It's there to be a base class to the class that contains all the properties that will _actually_ get serialized.  
Just one way to use this new feature.

Reply:
Then perhaps it should not be titled Config**File**

Reply:
```cs
public class MyConfig : ConfigFile
```
`MyConfig` is a configuration in this case, not a configuration **file**

Reply:
The entire point of this class is to facilitate serialization of this class to & from a json _file_.